### PR TITLE
Feature/multiinstance refactor

### PR DESCRIPTION
**Pull request by essweine**
This is a complete refactor of how we handle multiinstance tasks.  We no longer:
- use dynamically generated classes
- copy tasks and insert them into the task tree manually during `predict`
- modify the workflow spec while the workflow is in progress (except in the Camunda version)
- rely on Camnuda extensions in the base package

During parsing, a multiinstance task is added to the spec, with a reference to the original task spec.  When the MI task is reached, it changes state to waiting and creates children from the original referenced spec as needed, merges the data from the children into the original task as they are completed, and when all children are complete, moves to ready/complete.

This GREATLY simplifies dealing with multiinstance tasks.

### CODE REVIEW

File: SpiffWorkflow/bpmn/serializer/migration/version_1_1.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                subprocesses[sp]['tasks'][task_id] = dct['tasks'].pop(task_id)
+            if subprocesses[sp]['root'] is None:
+                subprocesses[sp]['root'] = task_id
+                subprocesses[sp]['tasks'][task_id]['parent'] = None
+
+    # Fix up th task and workflow states
+    waiting = []
+    for sp in subprocesses:
+        completed = sorted(
+            [t for t in subprocesses[sp]['tasks'].values() if t['state'] in [32, 64] ],
```
Comment (jbirddog) [inline SpiffWorkflow/bpmn/serializer/migration/version_1_1.py:17]: Safer/more obvious to use the finished mask?

File: SpiffWorkflow/bpmn/serializer/migration/version_1_1.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        # If there are uncompleted tasks, set the subworkflow task state to waiting
+        if len(completed) < len(subprocesses[sp]['tasks']):
+            waiting.append(sp)
+
+    # Check the top level and all subprocesses for waiting tasks
+    # Also remove any children that are no longer in the tree
+    for sp in [dct] + list(subprocesses.values()):
+        for task_id, task in sp['tasks'].items():
+            if task_id in waiting:
+                task['state'] = 8
```
Comment (jbirddog) [inline SpiffWorkflow/bpmn/serializer/migration/version_1_1.py:31]: Same thing-ish here - TaskState.WAITING?

**Review by jbirddog — APPROVED** [review]
Minor comments, lratm (looks really awesome to me).

File: SpiffWorkflow/bpmn/serializer/migration/version_1_1.py (hunk 1 above)
Comment (essweine) [reply SpiffWorkflow/bpmn/serializer/migration/version_1_1.py:17]: I am going to leave these as ints, since that's what's in the serialied version.

**Review by burnettk — APPROVED** [review]

**Pull request merged**